optimizely/client/factory.go

```diff
@@ -18,25 +18,34 @@ package client
 
 import (
 	"context"
+	"errors"
 	"fmt"
 
 	"github.com/optimizely/go-sdk/optimizely"
 	"github.com/optimizely/go-sdk/optimizely/config"
 	"github.com/optimizely/go-sdk/optimizely/decision"
 )
 
+const datafileURLTemplate = "https://cdn.optimizely.com/datafiles/%s.json"
```

Comment:
I have code stashed that moves this out of the client factory. Will put up in a separate PR.